File: src/types/result.ts
export interface QueryResult {
  queryId: string;
  queryText: string;
  provider: string;
  model: string;
  response: string;
  latencyMs: number;
  tokenUsage?: TokenUsage;
  iteration: number;
  timestamp: string;
  error?: string;
}

export interface TokenUsage {
  promptTokens: number;
  completionTokens: number;
  totalTokens: number;
}

export interface ExecutionRun {
  runId: string;
  brand: string;
  providers: string[];
  totalQueries: number;
  completedQueries: number;
  failedQueries: number;
  results: QueryResult[];
  startedAt: string;
  completedAt?: string;
  status: 'running' | 'completed' | 'failed' | 'partial';
}

File: src/voyage_geo/types/result.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

from voyage_geo.storage.schema import SCHEMA_VERSION


class TokenUsage(BaseModel):
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int


class QueryResult(BaseModel):
    query_id: str
    query_text: str
    provider: str
    model: str
    response: str
    latency_ms: int
    token_usage: TokenUsage | None = None
    iteration: int = 1
    timestamp: str = ""
    error: str | None = None


class ExecutionRun(BaseModel):
    schema_version: str = SCHEMA_VERSION
    run_id: str
    brand: str
    providers: list[str]
    total_queries: int
    completed_queries: int = 0
    failed_queries: int = 0
    results: list[QueryResult] = []
    started_at: str = ""
    completed_at: str | None = None
    status: Literal["running", "completed", "failed", "partial"] = "running"
